tasks/configure_filesystems.yml

```diff
@@ -1,66 +1,66 @@
 ---
-- name: Gather zfs filesystem attributes for {{ parent_fs }}/{{ fs.name }}
+- name: Gather zfs filesystem attributes for {{ zfs_parent_fs }}/{{ fs.name }}
   zfs_facts:
-    name: "{{ parent_fs }}/{{ fs.name }}"
+    name: "{{ zfs_parent_fs }}/{{ fs.name }}"
     properties: "utf8only,normalization,casesensitivity"
-  ignore_errors: True
+  failed_when: False
   register: fs_exists
 
 # this task fails when the filesystem already exists and some attribute in {utf8only, normalization, casesensitivity} is already set and has a differing value than the variables dictate. This is done because these attributes cannot be changed after creation
-- name: Fail if readonly attributes would be changed
+- name: Fail if readonly attributes would be changed on {{ zfs_parent_fs }}/{{ fs.name }}
   fail: 
-    msg: "tried to change readonly zfs attribute on already-existing filesystem {{ parent_fs }}/{{ fs.name }}\n {{ansible_zfs_datasets[0]}}"
+    msg: "tried to change readonly zfs attribute on already-existing filesystem {{ zfs_parent_fs }}/{{ fs.name }}\n {{ansible_zfs_datasets[0]}}"
   when: >
-    (fs_exists|succeeded) and
+    (fs_exists.msg is not defined) and
     (   ((fs.attributes is defined and fs.attributes.utf8only is defined or default is defined and default.utf8only is defined) and ansible_zfs_datasets[0].utf8only != fs.attributes.utf8only |default(zfs_storage_defaults.utf8only) |default(ansible_zfs_datasets[0].utf8only)) or
```

Comment:
No. `zfs_facts` fails when a filesystem doesn't exist. At least the ugliness of unnecessary failed tasks is avoided this way. 
Another way I can think of is executing and parsing a `zfs` command, but I think the level of ugliness of that exceeds the one of the current implementation.